Documents/AnalyticsToolDesignDocument.md

```diff
@@ -0,0 +1,48 @@
+# VFS For Git: Analytics Tooling Design Document
+
+This document details the design of and plan for the **VFS4G analytics tool**. The tool aims to help users identify reasons why their enlistment may be slowing down and the `gvfs` commands running slowly. By collecting and reporting statistics, the power to uncover and hopefully fix problems as they arrive moves to the user, empowering them to use the software to its fullest potential.
+
+## Problem
+
+When users unintentionally hydrate too many files within their enlistment, commands like checkout and commit increase their running time by several orders of magnitude. This causes unacceptable losses in performance and greatly hinders users as they go about their daily development cycle. The primary metric of the overhydration is the number of placeholder files which have been pulled from the remote into the local system. Slowdowns may also stem from having many modified files or directories which increases the amount of work that git does under the hood. Additionally, if a local enlistment has been in use for a long time, the number of loose object may eventually reach the order of several thousand in which case that also contributes to the loss in performance.
+
+## Provided Statistics
+
+### Safe to access
+
+- **Modified paths** - The number of files which have been edited and have been handed over to git
+- **Git index** - The differences between modified paths and the index tracked by git
+- **Placeholders** - A total number of the placeholders being tracked in VFS4G and a list of the placeholders in the system (Note that this contains some upper level directories and thus not all of their children are included in the count without enumeration)
+- **Troublesome programs** - Similar to what is already being logged in the heartbeat, look at which services or programs are responsible for hydrating large amounts of files in the enlistment
+
+### Potentially detrimental to access on a mounted enlistment
+
+- **Placeholder details** - Crawling the file system to enumerate directories of placeholders may unintentionally further hydrate files (IE downloading file sizes), but doing so would allow for greater details such as the percent hydration of folders, the total number of hydrated files, the type of files which are most hydrated, etc.
+- **File system** - Provide enumerated details for files in the placeholder list or the modified path list

```

Comment:
Anything in the modified paths list should be okay to get details on since they has been written to and there will not be any events happening inside of VFSForGit on them.

Reply:
This has since been removed and updated 😄